src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/config/Config.java

```diff
@@ -410,9 +421,41 @@ public void setValues(JSONObject formData) {
             useRestApi = false;
         }
 
+        filterIn = getFilterInFromFormData(formData);
+        updateServerEventFilter(formData.optString("name"));
+
         replicationConfig = ReplicationConfig.createReplicationConfigFromJSON(formData);
     }
 
+    /**
+     * Get the filter event list from formdata.
+     * @param formData JSONObject
+     * @return value.
+     */
+    private List<String> getFilterInFromFormData(JSONObject formData) {
+        List<String> filter = new ArrayList<>();
+        String[] arrayIn = new String[0];
+        String stringIn = formData.optString("filterIn");
+        if (stringIn.length() > 0) {
+            arrayIn = stringIn.substring(1, stringIn.length() - 1).split(", ");
```

Comment:
It looks like you're trying to remove the first and last character? Could you explain why this is necessary?

Reply:
The string looks something like this: "[one, two, three]". So I remove the first and last character before I split. I don't really need the first and last character though when I create the string. I just made it like that for uniformity. When a List<String> is sent to a text field it is format in the same way. I dont know if you can turn a string like that directly back into a List<String> though.

Reply:
Looks like it's a JSON array (assuming I'm reading it correctly)? You it would be better off to use `getJSONArray` instead of parsing yourself. https://stackoverflow.com/a/22074360/2573646

Reply:
A json array wraps strings around quotation marks. I could add this in order to fetch it as a jsonarray or remove the brackets to skip the substring part.

Reply:
Ok so trying to get a json array from the form before it exists will throw an exception. Plus I still want  to handle it as a List so I'd need to convert the JSONArray as well. You could handle the exception and keep it as a JSONArray but the check contains you need to do a toString on the array. No solution is perfect.

Reply:
That's fair. I figured since the JSON was already parsed, it would make more sense to use the pre-parsed version (especially if the input is big). If there's no way to get an empty list without an exception, it's probably fine either way. I'd just leave a comment near the code to explain what you're doing (and why).